test/functional/test_framework/script.py

```diff
@@ -872,7 +872,7 @@ def taproot_tree_helper(scripts):
 # - merklebranch: the merkle branch to use for this leaf (32*N bytes)
 TaprootLeafInfo = namedtuple("TaprootLeafInfo", "script,version,merklebranch,leaf_hash")
 
-def taproot_construct(pubkey, scripts=None):
+def taproot_construct(pubkey, scripts=None, treat_internal_as_infinity=False):
```

Comment:
needs to be described in below comments

Reply:
> if they don't verify whether or not a revealed internal public key in a script path spend is valid, and instead implement output_key == tweak(internal_key, tweakval) but such that tweak(invalid_key, tweakval) equals the public key corresponding to private key tweakval

Ok so you're saying an implementation *could*, instead of fail detecting it's not on the curve, could instead just consider the pubkey as the point at infinity, which would result in the tweak alone being the key, and in fact this may be a common mistake.